service/src/main/java/bio/terra/drshub/util/RequestUtils.java

```diff
@@ -0,0 +1,17 @@
+package bio.terra.drshub.util;
+
+import bio.terra.drshub.generated.model.ServiceName;
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.Optional;
+
+public class RequestUtils {
+
+  public static Optional<ServiceName> serviceNameFromRequest(HttpServletRequest request) {
+    var header = Optional.ofNullable(request.getHeader("x-terra-service-id"));
```

Comment:
I think it should be `X-Terra-Service-ID`?

Reply:
headers are case-insensitive! This is tested in unit tests.

Reply:
oh it is case insensitive